File: backend/node/src/exchange-rate/entities/exchange-rate.ts
import { Entity, Column, PrimaryGeneratedColumn, Index } from 'typeorm';

@Entity()
export class ExchangeRate {
  @PrimaryGeneratedColumn()
  Id: number;

  @Column({ type: 'date' })
  Date: Date;

  @Index()
  @Column({ type: 'date' })
  CreatedAt: Date;

  @Column({ type: 'char', length: 3 })
  Abbreviation: string;

  @Column({ type: 'decimal', precision: 14, scale: 12 })
  Rate: number;
}

File: backend/_others_/cs_dapper/Models/ExchangeRate.cs
namespace cssharp.Models
{
  public class ExchangeRate
  {
    public int Id { get; set; }

    public DateTime Date { get; set; }

    public DateTime CreatedAt { get; set; }

    public string Abbreviation { get; set; } = "";

    public decimal Rate { get; set; }
  }
}
